libsolidity/interface/CompilerStack.cpp

```diff
@@ -1611,6 +1618,11 @@ std::string CompilerStack::createMetadata(Contract const& _contract, bool _forIR
 	case CompilationSourceType::SolidityAST:
 		sourceType = "SolidityAST";
 		break;
+	case CompilationSourceType::EvmAssemblyJSON:
+		sourceType = "EvmAssemblyJson";
+		break;
```

Comment:
> We should remove `CompilationSourceType::EvmAssemblyJSON`. I think this is the last remaining place where it's still used for anything, and honestly, it never made sense here.

I remove it :)

> It being here is a left-over from previous state of this PR where the import was hacked into `CompilerStack` and had to handle various things that are relevant only to compilation from source. When we're importing assembly, we never create metadata on our own. We instead get a binary blob in `.auxdata`, which is the CBOR data containing metadata hashes.
> 
> Maybe it would be good to be able to tag it as created from imported assembly (as we do when we import the AST), but that's impossible without access to the original metadata.

But wouldn't be possible to detect if it is a binary blob and then tag it? Or you mean that the tag itself would be stored as metadata, thus we would not be able to append such information.

Reply:
I meant that we can't add that info to metadata because the blob does not contain metadata. It's the CBOR stub that only has the hashes of the metadata JSON. If we decode it, we can put some stuff next to the hashes but not inside the JSON. I don't think this is acceptable because someone viewing the metadata would not see that information. It would only be directly in the binary and not human-readable, which is the whole point of metadata.

And an extra obstacle is that `.auxdata` might not even be the CBOR blob. In the assembly the compiler generates it is, but if we, get it from the outside, it could be replaced with anything.